jib-core/src/main/java/com/google/cloud/tools/jib/configuration/credentials/CredentialRetriever.java

```diff
@@ -27,7 +27,9 @@
    *
    * @return the fetched credentials or {@code null} if no credentials could be fetched with this
    *     provider
+   * @throws Exception if the execution should fail instead of trying other credential retrieval
+   *     methods
    */
   @Nullable
-  Credentials retrieve();
+  Credentials retrieve() throws Exception;
```

Comment:
I think you should introduce a specific exception.

Reply:
Something like a `CredentialRetrievalException`?